posts/blundell/You_have_permission_to.jade

```diff
@@ -0,0 +1,204 @@
+                at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
+                at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
+                at android.app.ActivityThread.-wrap11(ActivityThread.java)
+                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
+                at android.os.Handler.dispatchMessage(Handler.java:102)
+                at android.os.Looper.loop(Looper.java:148)
+                at android.app.ActivityThread.main(ActivityThread.java:5417)
+                at java.lang.reflect.Method.invoke(Native Method)
+                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
+                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
+        Caused by: java.lang.SecurityException: Permission Denial: opening provider com.android.providers.contacts.ContactsProvider2 from ProcessRecord{cbb85e8 19360:com.blundell.contactscrash/u0a143} (pid=19360, uid=10143) requires android.permission.READ_CONTACTS or android.permission.WRITE_CONTACTS
+                at android.os.Parcel.readException(Parcel.java:1599)
+                at android.os.Parcel.readException(Parcel.java:1552)
+                at android.app.ActivityManagerProxy.getContentProvider(ActivityManagerNative.java:3550)
+                at android.app.ActivityThread.acquireProvider(ActivityThread.java:4778)
+                at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2018)
+                at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1468)
+                at android.content.ContentResolver.query(ContentResolver.java:475)
+                at android.content.ContentResolver.query(ContentResolver.java:434)
+                at android.provider.ContactsContract$Contacts.getLookupUri(ContactsContract.java:1500)
+                at com.blundell.contactscrash.MainActivity.onCreate(MainActivity.java:13)
+    ```
+
+    Ok ok this seems to be telling us what we want to hear, this is how I'd go about reading and thinking through this stacktrace:
+
+    `12-02 07:36:13.467` Yep this is the correct crash, I ran the app at 7:36am this morning ... _I know right_.
+    `Process: com.blundell.contactscrash, PID: 19360` Yep its my app that is crashing.
+    `java.lang.RuntimeException: Unable to start activity ComponentInfo{com.blundell.contactscrash/com.blundell.contactscrash.MainActivity}` Ok so we crashed on start up (MainActivity: onCreate failed to complete)
+    `java.lang.SecurityException: Permission Denial: opening provider com.android.providers.contacts.ContactsProvider2` Permission denial?? But I have the permission right _thur_ in my AndroidManifest
```

Comment:
`thur`? `there` or `therrrrrre` if trying to emphasise with a silly voice :tongue: